spec/message.ebnf

```diff
@@ -1,25 +1,20 @@
-Message ::= Declaration* ( Pattern | Selector Variant+ )
+Message ::= (s? Declaration)* s? Body s?
 
-/* Preamble */
-Declaration ::= 'let' WhiteSpace Variable '=' '{' Expression '}'
-Selector ::= 'match' ( '{' Expression '}' )+
+Declaration ::= 'let' s Variable s? '=' s? '{' s? Expression s? '}'
+Body ::= Pattern | Selector (s? Variant)+
 
-/* Variants and Patterns */
-Variant ::= 'when' ( WhiteSpace VariantKey )+ Pattern
-VariantKey ::= Literal | Nmtoken | '*'
-Pattern ::= '{' (Text | Placeholder)* '}' /* ws: explicit */
+Pattern ::= '{' (Text | Placeholder)* '}'
+Selector ::= 'match' (s? '{' s? Expression s? '}')+
+Variant ::= 'when' (s Key)+ s? Pattern
+Key ::= Nmtoken | Literal | '*'
 
-/* Placeholders */
-Placeholder ::= '{' (Expression | Markup | MarkupEnd)? '}'
+Placeholder ::= '{' s? (Expression | Markup | MarkupEnd)? s? '}'
 
-/* Expressions */
-Expression ::= Operand Annotation? | Annotation
-Operand ::= Literal | Variable
-Annotation ::= Function Option*
-Option ::= Name '=' (Literal | Nmtoken | Variable)
+Expression ::= (Literal | Variable) (s Annotation)? | Annotation
```

Comment:
Is it intentional to drop out the "operand" term here?

Reply:
Yes, I didn't feel the need for the indirection it introduced. As I expressed elsewhere, I don't think the BNF has to directly map to the AST.